### DISCUSSION TITLE
Could not receive message

### DISCUSSION THREAD

**Discussion by Aendrew87** [AsamK/signal-cli#442] [Q&A] [answered]
I am able to send messages using `signal-cli -u <linux_number> send -m "This is a message" <android_number>`-
On my android phone I receive that message and send a reply to it.

On linux (debian) I tried to the receive the message from my android device using `signal-cli -u <linux_number> receive`.
This results in no error and no console output.
I also tried `signal-cli --verbose -u <linux_number> receive` which results in:
> 2021-01-17T17:16:59.764+0100 [main] INFO LibSignal - [WebSocketConnection]: connect()
> 2021-01-17T17:17:00.440+0100 [OkHttp https://textsecure-service.whispersystems.org/...] INFO LibSignal - [WebSocketConnection]: onOpen() connected
> 2021-01-17T17:17:04.800+0100 [main] INFO LibSignal - [WebSocketConnection]: disconnect()

My signal-cli version is 0.7.3.
Could someone help me out, please?

**Comment by PimpMySocial** [comment]
Same with Ubuntu 20.04 and iOs.
On the iOs app, status of message is sent.

**Comment by atticus-sullivan** [comment]
I'm having the same trouble here. The interesting thing is that I have two numbers registered on signal-cli. One is only linked to the registration on another device, the other number is really registered on signal-cli.

On the number which simply is linked receiving works, on the other one (the one fully registered with signal-cli) receiving does not work.

**Reply by atticus-sullivan** [reply]
Oh right, sending does of course work (from both numbers)

**Comment by pumbaa666** [comment]
Same problem !
Can't read from single user or group and can't write on groups either.

Here is a snippet to reproduce quickly on a debian environment.

**Setup**
```
USERNAME=+41779998877
RECIPIENT=+41771112233

#Register
signal-cli -u $USERNAME register
CODE=123-456
signal-cli -u $USERNAME verify $CODE

#Update profile to allow groups creation
signal-cli -u $USERNAME updateProfile --name "My name" # --remove-avatar
```

**To user**
```
MESSAGE="This is a message"
signal-cli -u $USERNAME send -m "$MESSAGE" $RECIPIENT
# OK

signal-cli -u $USERNAME -o json receive --ignore-attachments -t 10
# Ends with no result after timeout. Even without -o, -t or --ignore-attachments parameters
```

**To groups**
```
GROUP_NAME="This is a group"
RESULT=$(signal-cli -u $USERNAME updateGroup -n "$GROUP_NAME" -m $RECIPIENT)
GROUP_ID=$(echo "$RESULT" | sed -e 's/[^\"]*\"//' -e 's/\".*//')
signal-cli -u $USERNAME send -m "$MESSAGE" -g $GROUP_ID
# Get an id but no message is sent
```

**VERSIONS**
```
java --version
openjdk 11.0.6 2020-01-14
OpenJDK Runtime Environment (build 11.0.6+10-post-Debian-1bpo91)
OpenJDK 64-Bit Server VM (build 11.0.6+10-post-Debian-1bpo91, mixed mode, sharing)

signal-cli --version
signal-cli 0.7.3
```

**Reply by pumbaa666** [reply]
Same behavior with versions : 
- 0.7.3
- 0.7.2
- 0.7.0
- 0.6.12

**Comment by phoenix-anasazi** [comment]
Same problem with self-built on Raspberry Pi 4.
signal-cli 0.7.3

**Comment by AsamK** [comment, marked as answer]
There was an issue with saving the signalingKey after registration.
I've pushed a fix to master, will release a new version soon.

**Comment by AKwoKWH** [comment]
If the version 0.7.4 fixed the issue?

**Reply by AsamK** [reply]
yes, the issue is fixed in 0.7.4

**Comment by PimpMySocial** [comment]
Thank you for your work!

**Comment by Aendrew87** [comment]
thanks, 0.7.4 works great for me
